src/app/components/mno-product-selector/mno-product-selector.controller.coffee

```diff
@@ -10,19 +10,46 @@
     dismiss: '&'
   },
   templateUrl: 'app/components/mno-product-selector/mno-product-selector.html',
-  controller: ($window, orderByFilter, MnoeProducts) ->
+  controller: ($window, orderByFilter, MnoeProducts, MnoeApps) ->
     'ngInject'
 
     $ctrl = this
 
     $ctrl.$onInit = ->
-      $ctrl.products = orderByFilter($ctrl.resolve.products.data, 'name')
+      $ctrl.flag = $ctrl.resolve.dataFlag
+      $ctrl.resolveProducts()
       $ctrl.multiple = $ctrl.resolve.multiple
       $ctrl.modalHeight = ($window.innerHeight - 200) + "px"
       $ctrl.selectedProducts = []
       $ctrl.headerText = $ctrl.resolve.headerText || 'mnoe_admin_panel.components.mno-product-selector.title'
       $ctrl.actionButtonText = $ctrl.resolve.actionButtonText || 'mnoe_admin_panel.components.mno-product-selector.create_order'
 
+    $ctrl.resolveProducts = ->
+      params = {
+        skip_dependencies: true,
+        fields: {
+          products: ['name, logo']
+        }
+      }
+      $ctrl.isLoadingProducts = true
+
+      if $ctrl.flag == 'organization-create-order'
```

Comment:
I'd use a `switch/case` here instead of multiple `if`

You could also factor the `finally` and the `$ctr.products` that are repeated.
So the `switch` could just set a promise that you acts on after to DRY the code